wagtail_transfer/static_src/components/ContentImportForm/index.js

```diff
@@ -68,16 +88,34 @@ export default function ImportContentForm({
     if (!sourcePage && destPage) {
       setDestPage(null);
     }
-  }, [source, sourcePage, destPage]);
+  }, [source, sourcePage, destPage, sourceModel, sourceModelObjectId]);
+
+  React.useEffect(() => {
+    if(sourceModel && !sourceModelObjectId) {
+      // There is a sourceModel but no sourceModelObjectId
+      const fetchNumPages = async () => {
+        const api = new ModelsAPI(source.page_chooser_api).query()
+        const page = await api.getModel(sourceModel.label)
+        setNumPages(page.meta.total_count);
```

Comment:
In this case, we're not importing pages are we? Could we make this name more generic?

Reply:
This one gets confusing. We're looking for pagination pages. The name `page` conflicts with a Wagtail Page so I've made this more generic to hopefully reduce confusion here.